src/main/groovy/6importCredentials.groovy

```diff
@@ -0,0 +1,232 @@
+}
+
+List credentialConfig = yaml.load(configText)
+credentialConfig.each { newCredential ->
+    credentialType = newCredential.credentialType
+    id = newCredential.id
+    description = newCredential.description
+    scopeString = newCredential.scope
+
+    // All credentials must have a credentialType and scope
+    if (!credentialType || !scopeString) {
+        logger.severe("Missing data for credential. Please ensure every entry in the " +
+                      "credentials.yml file has a credentialType and scope")
+        jenkins.doSafeExit(null)
+        System.exit(1)
+    }
+
+    switch (scopeString.toLowerCase()) {
+        case 'global':
+            scope = CredentialsScope.GLOBAL
+            break
+        case 'system':
+            scope = CredentialsScope.SYSTEM
+            break
+        case 'user':
+            scope = CredentialsScope.USER
+            break
+        default:
+            logger.severe("Invalid scope specified for credential. Scope " +
+                          "should be GLOBAL, SYSTEM, or USER. Was: ${scopeString}")
+            jenkins.doSafeExit(null)
+            System.exit(1)
+    }
+
+    switch (credentialType) {
+        case 'usernamePassword':
+            username = newCredential.username
+            password = newCredential.password
+
+            // Make sure the username and password fields are not empty
+            if (!username || !password) {
+                logger.severe("Missing data for credential. Please ensure usernamePassword " +
+                              "entries in the credentials.yml file have a " +
+                              "username and password")
+                jenkins.doSafeExit(null)
+                System.exit(1)
+            }
+
+            cred = new UsernamePasswordCredentialsImpl(
+                scope,
+                id,
+                description,
+                username,
+                password
+            )
+            credentialsStore.addCredentials(
+                globalDomain,
+                cred
+            )
+            break
+        case 'secretText':
+            text = newCredential.secretText
+
+            // Make sure the text field is not empty
+            if (!text) {
+                logger.severe("Missing data for credential. Please ensure secretText " +
+                              "entries in the credentials.yml file have text")
+                jenkins.doSafeExit(null)
+                System.exit(1)
+            }
+
+            secretText = new StringCredentialsImpl(
+                scope,
+                id,
+                description,
+                Secret.fromString(text)
+            )
+            credentialsStore.addCredentials(globalDomain, secretText)
+            break
+        case 'secretFile':
+            path = newCredential.path
+            fileName = newCredential.name
+
+            // Make sure the path and name is not empty
+            if (!path || !fileName) {
```

Comment:
I think you should also check that the file exists, or wrap the new file + copy in a try/catch